src/Avalonia.Controls/DrawingPresenter.cs

```diff
@@ -0,0 +1,58 @@
+﻿using Avalonia.Controls.Shapes;
+using Avalonia.Media;
+using Avalonia.Metadata;
+
+namespace Avalonia.Controls
+{
+    public class DrawingPresenter : Control
+    {
+        public static readonly StyledProperty<Drawing> DrawingProperty =
+            AvaloniaProperty.Register<DrawingPresenter, Drawing>(nameof(Drawing));
+
+        [Content]
+        public Drawing Drawing
+        {
+            get => GetValue(DrawingProperty);
+            set => SetValue(DrawingProperty, value);
+        }
+
+        public static readonly StyledProperty<Stretch> StretchProperty =
+            AvaloniaProperty.Register<DrawingPresenter, Stretch>(nameof(Stretch), Stretch.Uniform);
+
+        public Stretch Stretch
+        {
+            get => GetValue(StretchProperty);
+            set => SetValue(StretchProperty, value);
+        }
+
+        static DrawingPresenter()
+        {
+            AffectsMeasure(DrawingProperty);
+            AffectsRender(DrawingProperty);
+        }
+
+        private Matrix _transform = Matrix.Identity;
+
+        protected override Size MeasureOverride(Size availableSize)
+        {
+            if (Drawing == null) return new Size();
+
+            var (size, transform) = Shape.CalculateSizeAndTransform(availableSize, Drawing.GetBounds(), Stretch);
+
+            _transform = transform;
+
+            return size;
+        }
+
+        public override void Render(DrawingContext context)
+        {
+            if (Drawing != null)
+            {
+                using (context.PushPreTransform(_transform))
```

Comment:
I think to address the issue of required clipping in `UniformToFill` you could just push a clip here?